### ISSUE TITLE
The WPF window does not look the same in design mode as it does when built

### ISSUE THREAD

**Issue by Grey-Wind** [dotnet/wpf#9725]
### Description

The WPF window does not look the same in design mode as it does when built.

The window style is `SingleBorderWindow `or `ThreeDBorderWindow`.

I don't know if this will be a problem in `ToolWindow `style

### Reproduction Steps

Create a WPF project and build.

You can add a test label.

### Expected behavior

The contents on the right and bottom are displayed completely.

### Actual behavior

The content on the right and below is incomplete or cropped.

### Regression?

I didn't test the older version.

### Known Workarounds

Use the `None` window style.

### Configuration

.net6 to .net8

windows10 22h2

any cpu

### Other information

[Video_CN](https://www.bilibili.com/video/BV1BtpMeKEAf/)

[Video_EN](https://www.youtube.com/watch?v=MSsYZjcjtis)

The project in the video is .net8

**Comment by miloush** [comment]
As far as I can see everything seems to be working as expected. The `Width` and `Height` properties on `Window` include the borders and caption/title bar. That's why `None` window style meets your expectations, because the content area is not decreased by them. When you set the same dimensions on the child, it does not fit the window and will be cropped, exactly how the designer shows you. You can use `SizeToContent=true` to resize the window automatically to fit all of the content.

Are you simply trying to align the label to the right bottom corner? Why use the `Margin` instead of the `HorizontalAlignment` and `VerticalAlignment`?

**Comment by Grey-Wind** [comment]
I am used to using margin in my daily life. Since there is no need to develop very large software, margin does not affect normal development except for this display problem.

**Comment by Grey-Wind** [comment]
The problem persists, and I'll post another demo video when I have time later.

**Comment by Grey-Wind** [comment]
https://github.com/user-attachments/assets/65f5f7dc-9cbe-47be-9ee9-e7aa364e8391

**Comment by Grey-Wind** [comment]
As shown in the video, I reproduced the problem again in Visual Studio 17.14.11.

My code is as follows. I don't think there is any problem.

```
<Window
    x:Class="Imger.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:local="clr-namespace:Imger"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    x:Name="MainW"
    Title="Imger"
    Width="800"
    Height="450"
    ResizeMode="CanMinimize"
    mc:Ignorable="d">
    <Grid>
        <!--#region Grid Definitions-->
        <Grid.ColumnDefinitions>
            <ColumnDefinition Width="1*" />
            <ColumnDefinition Width="1*" />
        </Grid.ColumnDefinitions>
        <Grid.RowDefinitions>
            <RowDefinition Height="2*" />
            <RowDefinition Height="2*" />
        </Grid.RowDefinitions>
        <!--#endregion-->

        <!--#region Left-->
        <Label
            Grid.Row="0"
            Grid.RowSpan="3"
            Grid.Column="2"
            Grid.ColumnSpan="7"
            Content="Imger" />
        <Frame
            x:Name="OptimizationControlPanelFrame"
            Grid.Row="4"
            Grid.RowSpan="50"
            Grid.Column="0"
            Grid.ColumnSpan="60"
            NavigationUIVisibility="Hidden" />
        <!--#endregion-->

        <!--#region Right-->
        <Image
            x:Name="PreviewRendering"
            Grid.Row="0"
            Grid.RowSpan="50"
            Grid.Column="60"
            Grid.ColumnSpan="40"
            AllowDrop="True"
            DragDrop.Drop="PreviewRendering_Drop" />
        <!--#endregion-->
    </Grid>
</Window>
```

**Comment by Grey-Wind** [comment]
> 6b9c24d0fdde480a0118dbadc95c161a.mp4

As shown in the video, the Grid is not within the window but extends beyond the bottom and the right side.

**Comment by Grey-Wind** [comment]
I think this bug may appear directly from .NET 6 to .NET 9.

**Comment by Grey-Wind** [comment]
> I think this bug may appear directly from .NET 6 to .NET 9.

I didn't test older or newer versions.

**Comment by lindexi** [comment]
It seems the UseLayoutRounding issues.

**Comment by miloush** [comment]
@lindexi no, as I already noted above, there is no bug here. Width and Height include non-client area.
